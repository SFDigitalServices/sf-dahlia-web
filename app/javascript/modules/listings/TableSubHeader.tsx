import { t } from "@bloom-housing/ui-components"
import React from "react"

type TableSubHeaderProps = {
  priorityTypes: string[] | null
}

const TableSubHeader = ({ priorityTypes }: TableSubHeaderProps) => {
  return (
    priorityTypes && (
      <div>
        {t("listings.includesPriorityUnits")}
        <ul className="list-disc ml-4">
          {priorityTypes.map((name) => (
            <li key={name}>{name}</li>
          ))}
        </ul>
      </div>
    )
  )
}

export default TableSubHeader
