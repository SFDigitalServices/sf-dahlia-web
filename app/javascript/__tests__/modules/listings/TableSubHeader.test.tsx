import React from "react"
import renderer from "react-test-renderer"
import TableSubHeader from "../../../modules/listings/TableSubHeader"

describe("TableSubHeader", () => {
  it("renders TableSubHeader component", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })
    const tree = renderer.create(<TableSubHeader priorityTypes={["test"]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
