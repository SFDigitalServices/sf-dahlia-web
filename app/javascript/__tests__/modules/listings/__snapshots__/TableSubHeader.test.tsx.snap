// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSubHeader renders TableSubHeader component 1`] = `
<DocumentFragment>
  <div>
    Includes Priority Units for:
    <ul
      class="list-disc ml-4"
    >
      <li>
        test
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
